nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -97,6 +97,8 @@
 
 - `pass` now does not contain `password-store.el`.  Users should get `password-store.el` from Emacs lisp package set `emacs.pkgs.password-store`.
 
+- `services.knot` now supports `.settings` from RFC42.  The change is not 100% compatible with the previous `.extraConfig`.
```

Comment:
This should relay contain what is not compatible or some link to maybe here or somewhere else because right now lonely from this line I am totally unsure what to other than just give it a shot.

Reply:
I would also appreciate seeing more detail here. The error generated without making any changes to my previous config was (at least to my eyes) fairly cryptic:

<details>
<summary>Full error trace:</summary>

```
error:
       … while calling the 'head' builtin

         at /nix/store/8lg5579v1k5mc63ypzfagrcl8l32x3bh-source/lib/attrsets.nix:820:11:

          819|         || pred here (elemAt values 1) (head values) then
          820|           head values
             |           ^
          821|         else

       … while evaluating the attribute 'value'

         at /nix/store/8lg5579v1k5mc63ypzfagrcl8l32x3bh-source/lib/modules.nix:807:9:

          806|     in warnDeprecation opt //
          807|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          808|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: assertion '(((cfg).settings == { }) && ((cfg).keyFiles == [ ]))' failed

       at /nix/store/8lg5579v1k5mc63ypzfagrcl8l32x3bh-source/nixos/modules/services/networking/knot.nix:106:5:

          105|   configFile = if cfg.settingsFile != null then
          106|     assert cfg.settings == {} && cfg.keyFiles == [];
             |     ^
          107|     cfg.settingsFile
 ```
 </details>

Reply:
The error should be gone after merging PR https://github.com/NixOS/nixpkgs/pull/258361